### PR TITLE
Update samples to Visual Studio 2022

### PR DESCRIPTION
**Pull request by gregg-miskelly**
This commit updates all the samples in this repo to target Visual Studio 2022 (Version 17.0). More information on VS 2022 migration can be found in the [wiki](https://github.com/microsoft/ConcordExtensibilitySamples/wiki/Visual-Studio-2022-support).

### CODE REVIEW

File: build.cmd [hunk 1]
```diff
@@ -11,12 +11,12 @@ if NOT "%VSINSTALLDIR%"=="" goto InDevPrompt
...
 set VSWherePath=%x86ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe
-if NOT exist "%VSWherePath%" echo ERROR: Could not find vswhere.exe (%VSWherePath%). Ensure that Visual Studio 2017 version 15.5 or newer is installed. & exit /b -1 
+if NOT exist "%VSWherePath%" echo ERROR: Could not find vswhere.exe (%VSWherePath%). Ensure that Visual Studio 2022 is installed. & exit /b -1 
 
-for /f "usebackq tokens=1 delims=" %%a in (`"%VSWherePath%" -version [16.0, -prerelease -requires Microsoft.VisualStudio.Workload.NativeDesktop;Microsoft.VisualStudio.Workload.VisualStudioExtension;Microsoft.VisualStudio.Workload.ManagedDesktop -property installationPath`) do call :ProcessIDE "%%a"
+for /f "usebackq tokens=1 delims=" %%a in (`"%VSWherePath%" -version [17.0, -prerelease -requires Microsoft.VisualStudio.Workload.NativeDesktop;Microsoft.VisualStudio.Workload.VisualStudioExtension;Microsoft.VisualStudio.Workload.ManagedDesktop -property installationPath`) do call :ProcessIDE "%%a"
 if NOT "%VSINSTALLDIR%"=="" goto InDevPrompt
 
-echo ERROR: Unable to find a Visual Studio install.
+echo ERROR: Unable to find a Visual Studio 2022 or newer install.
```
Comment (plnelson) [inline build.cmd:19]: nit: you deleted "or newer" above.  I assume you don't want it here either?

File: HelloWorld/Cs/vsix/source.extension.vsixmanifest [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+        <DisplayName>C# HelloWorld Debugger Sample</DisplayName>
+        <Description xml:space="preserve">C# HelloWorld Debugger Sample</Description>
+    </Metadata>
+    <Installation>
+      <InstallationTarget Id="Microsoft.VisualStudio.Community" Version="[17.0, 18.0)">
+        <ProductArchitecture>amd64</ProductArchitecture>
+      </InstallationTarget>
+    </Installation>
+    <Prerequisites>
+        <Prerequisite Id="Microsoft.VisualStudio.Debugger" Version="[15.0,)" DisplayName="Visual Studio Debugger" />
```
Comment (plnelson) [inline HelloWorld/Cs/vsix/source.extension.vsixmanifest:14]: 17

File: build.cmd (hunk 1 above)
Comment (gregg-miskelly) [reply build.cmd:19]: Actually, I didn't mean to delete the 'or newer' above. I meant to just delete the fact that 15.5 was okay. Let me put the "or newer" back.

**Review by plnelson — APPROVED** [review]
👍

**Pull request merged**